lib/IRGen/GenClangDecl.cpp

```diff
@@ -82,6 +84,100 @@ clang::Decl *getDeclWithExecutableCode(clang::Decl *decl) {
   return nullptr;
 }
 
+auto walkCallGraphFromCtor(const clang::CXXConstructorDecl *toplevelCtor,
+                           std::set<const clang::Decl *> visited,
+                           std::vector<const clang::Decl *> &FDS) {
+  std::vector<clang::Stmt *> stack;
+
+  // Keep track of decls we have already walked over so that we don't re-walk
+  // over them redundantly.
+  auto visitDecl = [&visited](const clang::Decl *decl) {
+    const bool hasVisited = visited.find(decl) != visited.end();
+    if (!hasVisited) {
+      visited.insert(decl);
+    }
+    return hasVisited;
+  };
+
+  auto HandleCtor = [&stack, &FDS](const clang::CXXConstructorDecl *ctor) {
+    FDS.push_back(ctor);
+    stack.push_back(ctor->getBody());
+    for (auto *init : ctor->inits()) {
+      stack.push_back(init->getInit());
+      if (clang::FieldDecl *field = init->getMember())
+        stack.push_back(field->getInClassInitializer());
+    }
+  };
+
+  auto HandleFunctionDecl = [&stack, &FDS](clang::FunctionDecl *F) {
+    LLVM_DEBUG({
+      llvm::dbgs() << "HANDLE FUNCTION DECL:\n";
+      llvm::dbgs() << "IS INLINE " << (F->isInlineSpecified() ? "YES" : "NO")
+                   << "\n";
+      llvm::dbgs() << "IS TEMPLATE INST "
+                   << (F->isTemplateInstantiation() ? "YES" : "NO") << "\n";
+    });
+
+    if (F->isInlineSpecified() || F->isTemplateInstantiation())
+      FDS.push_back(F);
+    stack.push_back(F->getBody());
+  };
+
+  HandleCtor(toplevelCtor);
+
+  while (!stack.empty()) {
+    auto *back = stack.back();
+    stack.pop_back();
+    if (!back)
+      continue;
+
+    LLVM_DEBUG({
+      llvm::dbgs() << "\nCandidate:\n";
```

Comment:
Not super important, but we might want a bit more debug info here. It might be weird to just get a random debug line that says "Candidate: <decl>". People might wonder, "candidate for what"?